src/modules/fancyzones/lib/FancyZonesData.cpp

```diff
@@ -216,58 +210,56 @@ void FancyZonesData::CloneDeviceInfo(const std::wstring& source, const std::wstr
     deviceInfoMap[destination] = deviceInfoMap[source];
 }
 
-void FancyZonesData::UpdatePrimaryDesktopData(const std::wstring& desktopId)
+void FancyZonesData::UpdatePrimaryDesktopData(const GUID& desktopId)
 {
     // Explorer persists current virtual desktop identifier to registry on a per session basis,
     // but only after first virtual desktop switch happens. If the user hasn't switched virtual
     // desktops in this session value in registry will be empty and we will use default GUID in
     // that case (00000000-0000-0000-0000-000000000000).
     // This method will go through all our persisted data with default GUID and update it with
     // valid one.
-    auto replaceDesktopId = [&desktopId](const std::wstring& deviceId) {
-        return deviceId.substr(0, deviceId.rfind('_') + 1) + desktopId;
-    };
+
     std::scoped_lock lock{ dataLock };
     for (auto& [path, perDesktopData] : appZoneHistoryMap)
     {
         for (auto& data : perDesktopData)
         {
-            if (ExtractVirtualDesktopId(data.deviceId) == NonLocalizable::DefaultGuid)
+            if (data.deviceId.virtualDesktopId == GUID_NULL)
             {
-                data.deviceId = replaceDesktopId(data.deviceId);
+                data.deviceId.virtualDesktopId = desktopId;
             }
         }
     }
-    std::vector<std::wstring> toReplace{};
+    std::vector<FancyZonesDataTypes::DeviceIdData> toReplace{};
     for (const auto& [id, data] : deviceInfoMap)
     {
-        if (ExtractVirtualDesktopId(id) == NonLocalizable::DefaultGuid)
+        if (id.virtualDesktopId == GUID_NULL)
         {
             toReplace.push_back(id);
         }
     }
-    for (const auto& id : toReplace)
+    for (auto& id : toReplace)
```

Comment:
I think body within this loop can be extracted to a lambda / separate method, called like `ReplaceDesktopId`.